### DISCUSSION TITLE
Применение ярлыков в таблице циклов

### DISCUSSION THREAD

**Discussion by LostAlly** [totumonline/totum-ru-issues-and-discussions#472] [Как это сделать?] [answered]
Приветствую.
Можете попродробней с примерами разяснить в каких случаях необходимо/желательно/можно использовать ярлыки ссылающиеся на поле в хедере внутри цикла, а в каких случаях можно/нужно использовать обычное поле(не ярлык)
По сути результат они дают примерно одинаковый на первый взгляд.
Спасибо.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Ярлыки для лени сделаны — в случаях когда нужно поднять изнутри цикла в таблицу циклов штук 20-30 полей это очень помогает, тк ярлык тащит за собой тип поля и находится в полном slave-режиме относительно основного поля без писания кода.

Есть 2 случая, когда необходимо использовать поле другого типа и коды для этой задачи:

1. Поднятие Индивидуального Селекта.
     a. Ошибка «поле не найдено» если расчет селекта внутри зависит от поля, которого нет в таблице Циклов
     b. Нагрузка — тк внутри цикла он один, а когда в таблице Циклов их может быть 1000 на странице
     
    > В этих случаях в т. Циклов создается простой Селект, который кодом берет значение изнутри цикла (параметр cycle можно не указывать для такого select и других ф. Тотум автоматически подставляет #id в параметр cycle при исполнении функций строка в Циклов —> т. внутри цикла)

2. Когда нужно изменять поле не только внутри цикла, но и в таблице циклов. Вот здесь показано: https://ru.docs.totum.online/label-type-fields

**Comment by UPRLS9000** [comment]
А есть смысл создавать внутри Цикла поле типа Дата с массивом нужной информации и вытаскивать его наверх в таблицу Циклов или это лишние заморочки?

**Reply by ttm-alexeypolunin** [reply]
Это имеет смысл, когда достается какая-то агрегированная информация из строчной части внутри. Тогда в футере собирается техническое поле тип Данные и оно уже вытягивается ярлыком.

**Reply by UPRLS9000** [reply]
Спасибо
